go/private/actions/link.bzl

```diff
@@ -24,6 +24,30 @@ load(
     "LINKMODE_PLUGIN",
 )
 
+def _format_archive(d):
+    return "{}={}={}".format(d.label, d.importmap, d.file.path)
+
+def _map_archive(x):
```

Comment:
I'd rather keep this the way it was. `test_archives` will pretty much always be 0 or 1 elements, so the filtering using `any` before should actually be very fast (probably faster than building a `dict` and doing hash table lookups). I think it was shorter and more readable before, so I prefer that.

I like having `_format_archive` as a separate function called with `map_each` though.

Reply:
Ok, makes sense - I'll change it back to `any(...)`.
The main motivation to extract this code into a function it to postpone the cost (depset expansion and execution of this code) to the execution phase and, likely, not pay it at all if the output is not needed. See https://docs.bazel.build/versions/master/skylark/performance.html#use-ctx-actions-args-for-command-lines
Basically Args is a way to postpone and avoid the cost of constructing the arguments list. The more processing is delegated to it the better. Currently each instance of the rule where this function is used pays the cost. After this refactoring only the ones where the argument list is actually needed will pay the cost. Does it make sense?